core/src/test/scala/org/apache/spark/storage/StorageStatusListenerSuite.scala

```diff
@@ -150,4 +158,12 @@ class StorageStatusListenerSuite extends FunSuite {
     listener.onUnpersistRDD(SparkListenerUnpersistRDD(1))
     assert(listener.executorIdToStorageStatus("big").numBlocks === 0)
   }
+  
+  test("Killed Executor Entry removed after configurable time") {
+    val localtestconf = new SparkConf().set(StorageStatusListener.TIME_TO_EXPIRE_KILLED_EXECUTOR,"5s")
+    val listener = new StorageStatusListener(localtestconf)
+    listener.removedExecutorIdToStorageStatus.put("1", new StorageStatus(null, 50))
+    Thread.sleep(5500)
```

Comment:
you can avoid sleeping by using [`cache.setTicker`](https://github.com/google/guava/wiki/CachesExplained#testing-timed-eviction)